packages/rainbowkit/src/components/ConnectOptions/MobileOptions.tsx

```diff
@@ -44,10 +44,26 @@ function WalletButton({ wallet }: { wallet: WalletConnector }) {
 
         onConnecting?.(async () => {
           if (getMobileUri) {
-            window.location.href = await getMobileUri();
+            const mobileUri = await getMobileUri();
+
+            if (mobileUri.startsWith('http') && mobileUri.includes('?uri=')) {
+              window.localStorage.setItem(
+                'WALLETCONNECT_DEEPLINK_CHOICE',
+                JSON.stringify({
+                  href: mobileUri.split('?')[0],
+                  name,
+                })
+              );
+            }
+
+            if (mobileUri.startsWith('http')) {
+              window.open(mobileUri, '_blank', 'noreferrer,noopener');
```

Comment:
This was added in case the fallback website is presented to the user so that it doesn't take over the current tab. This mirrors the way the official WalletConnect modal works.